### ISSUE TITLE
Wrong type expected for **kwargs in function with optional positional parameter

### ISSUE THREAD

**Issue by ethantkoenig** [astral-sh/ty#2944]
### Summary

```python
def f(x: int = 0, **kwargs: object):
    pass

f(**{'foo':'bar'})
```

```
error[invalid-argument-type]: Argument to function `f` is incorrect
 --> example.py:4:3
  |
2 |     pass
3 |
4 | f(**{'foo':'bar'})
  |   ^^^^^^^^^^^^^^^ Expected `int`, found `Unknown | str`
  |
info: Element `str` of this union is not assignable to `int`
info: Function defined here
 --> example.py:1:5
  |
1 | def f(x: int = 0, **kwargs: object):
  |     ^ ---------- Parameter declared here
2 |     pass
  |
info: rule `invalid-argument-type` is enabled by default

Found 1 diagnostic
```

### Version

ty 0.0.20

**Comment by charliermarsh** [comment]
Do you mind sharing a real-world snippet where this came up? (I'm trying to understand if the request is limited to dict literals, or something more general.)

**Comment by carljm** [comment]
I think this makes sense. Unpacking is generally forgiving (we assume your list is of the right length, etc), so we should probably also assume you're not unpacking into optional arguments. #1584 is also precedent for being forgiving in these kinds of cases, and assuming you know what you're doing.

@charliermarsh I don't think this is specific to dict literals. The point here isn't the thing that's being unpacked (it could be whatever -- we validate that it's a mapping, but we may not know specifically what keys or values it has), it's that we shouldn't require its value type to be assignable to the type of an optional argument in the target (`x` in this case); instead we should forgivingly assume that the mapping you're unpacking just doesn't have that key.

Fixing this will also have some overlap with what @ibraheemdev is doing in https://github.com/astral-sh/ruff/pull/23436
